jdbc/src/main/java/org/openconnectors/source/QuerierThread.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors.source;
+
+import org.openconnectors.data.Record;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+public class QuerierThread extends Thread {
+    private static final Logger logger = LoggerFactory.getLogger(QuerierThread.class);
+    private TableQuerier tableQuerier;
+    private long queryPeriod;
+    private AtomicBoolean isActive = new AtomicBoolean(false);
+    private Consumer<Collection<Record>> consumeFunction;
+
+    public QuerierThread(TableQuerier tableQuerier, long queryPeriod, Consumer<Collection<Record>> consumeFunction) {
+        this.tableQuerier = tableQuerier;
+        this.queryPeriod = queryPeriod;
+        this.consumeFunction = consumeFunction;
+    }
+
+    @Override
+    public void run() {
+        isActive.set(true);
+        while (isActive.get()) {
```

Comment:
If I understand correctly, when using bulk Querier, we get all rows of the table. Why are we doing that again and again in an infinite loop?

Reply:
Yes, we get all rows. But it may be useful to make backups periodically, for example, every day. So, user can configure this period by setting up TABLE_CHECK_PERIOD. Also, I think it would be better to configure that parameter as following:
table.check.period = N seconds/minutes/hours/days 
or 
table.check.period = none  -  in such case connector would Querier works only once.

Reply:
One sample use case of using a jdbc connector might be to do a one time copy of items to some place else. Do you think it makes sense to add a flag to control the number of iterations?

Reply:
I think we need to define the supported usecases:
* one time full table dump
* regular full table dumps on a schedule - in this case I don't think we need to reimplement the full scheduler in openconnect, but rather accept a trigger to run the job from Airflow or another scheduler
* periodic incremental - for this we should look at something similar to bottled water: https://github.com/confluentinc/bottledwater-pg

Reply:
I agree that the full scheduler in openconnect is redundant, but I'd leave the feature of periodical running. This is easy to implement and it present in Kafka Connect, so I'd bet this feature is used by some devs. We can add an optional flag to config and if a user sets this flag, only then use periodical running.